src/cargo/util/errors.rs

```diff
@@ -286,6 +286,16 @@ impl From<Box<CargoError>> for CliError {
 }
 
 // =============================================================================
+// NetworkError trait
+
+pub trait NetworkError: Error + Send + 'static {
+    fn maybe_spurious(&self) -> bool { true }
```

Comment:
I think I put this on the wrong comment.  I dont think I want to bound the with_retry to the NetworkError trait. I think I want to check in with_retry if the error is NetworkError or not, but all the errors should all be CargoError bound?